security/advancedtls/pemfile_provider.go

```diff
@@ -113,41 +143,21 @@ func NewPEMFileProvider(o PEMFileProviderOptions) (*PEMFileProvider, error) {
 	identityTicker := time.NewTicker(o.IdentityInterval)
 	rootTicker := time.NewTicker(o.RootInterval)
 	ctx, cancel := context.WithCancel(context.Background())
-	// We pass a copy of PEMFileProviderOptions to the goroutine in case users
-	// change it after we start reloading.
+
 	go func() {
+		// Get the instant updates before the monitoring loop runs.
+		updateIdentityDistributor(provider.identityDistributor, o.CertFile, o.KeyFile)
```

Comment:
How about moving these calls to the top of the `for` loop, and inside the `select` when the ticker fires, just `break`.

Reply:
Sure, this looks better!
I also removed the `default` clause at the end of the `select` block, which will change it from a non-blocking select to a blocking select. I think blocking is generally better in our case, since we want to avoid useless spinning of the `for` loop.

Reply:
Thanks, I missed the `default` case.